lib/runner/test-runner/test-runner.js

```diff
@@ -42,50 +47,44 @@ module.exports = testRunner = {
 
     test.name = testPath.split('\\').pop().split('/').pop();
 
-        testRunner._testNumbers.hasOwnProperty(test.name) ?
-        test.number = testRunner._testNumbers[test.name] :
-        testRunner._testNumbers[test.name] = test.number = Object.keys(testRunner._testNumbers).length;
-
-        test.testPath = testPath;
-
-        testRunner._testsRunning++;
-
-        test.stderr.on('data', function(data) {
-          testRunner.emit('testRunner.childStderrReceived', data, test.number);
-        });
-
-        test.on('message', function(data) {
-          testRunner.emit('testRunner.testDataReceived', data, test.number);
-        });
-
-        test.on('close', function(exitCode) {
-          if (exitCode) {
-            if (testRunner._rerunCount === 0) {
-              process.exitCode = 1;
-            }
-            testRunner._testsToRerun.push(test.testPath);
-          }
-          testRunner._testsRunning--;
-          testRunner._testsRemaining--;
-          testRunner.emit('testRunner.testFinished', test.number, testRunner._rerunCount);
-          if (testRunner._testsRemaining === 0) {
-            if (testRunner._testsToRerun.length === 0 || testRunner._rerunCount === 0) {
-              testRunner.emit('testRunner.done');
-            } else {
-              testRunner.emit('testRunner.rerunTestsReadyToConfigure');
-            }
-          }
-        });
-
-        testRunner.emit('testRunner.testStarted', test.number, test.name);
-  },
-  _configureRerun() {
-    testRunner._rerunCount--;
-    testRunner._testFiles = testRunner._testsToRerun;
-    testRunner._testsToRerun = [];
-    testRunner._testsRemaining = testRunner._testFiles.length;
-    testRunner.emit('testRunner.configured');
+    testRunner._testNumbers.hasOwnProperty(test.name) ?
+      test.number = testRunner._testNumbers[test.name] :
+      testRunner._testNumbers[test.name] = test.number = Object.keys(testRunner._testNumbers).length;
+
+    test.testPath = testPath;
+
+    testRunner._testsRunning++;
+
+    test.stderr.on('data', function(data) {
+      testRunner.emit('testRunner.childStderrReceived', data, test.number);
+    });
+
+    test.on('message', function(data) {
+      testRunner.emit('testRunner.testDataReceived', data, test.number);
+    });
+
+    test.on('close', function(exitCode) {
+      if (exitCode) {
+        testRunner._testRerunMemo[test.testPath]--;
+        if (testRunner._testRerunMemo[test.testPath] >= 0) {
+          testRunner._testFiles.push(test.testPath);
+        }
+      } else {
+        testRunner._testsRemaining -= testRunner._testRerunMemo[test.testPath];
+        delete testRunner._testRerunMemo[test.testPath];
+      }
+      testRunner._testsRunning--;
+      testRunner._testsRemaining--;
```

Comment:
Is this decrementing too much? Since on line 73 `testRunner._testsRemaining` is being decreased?

Reply:
That definitely looks weird, but when running it, seems to be doing the right thing.
I think its effectively removing tests from the count that are no longer needed because they passed and reruns are no longer needed

Reply:
Ok, hmmm. Let's keep an eye on it then